app/controllers/organizations_controller.rb

```diff
@@ -1,24 +1,15 @@
 class OrganizationsController < ApplicationController
   after_action :verify_authorized
-  rescue_from Errno::ENAMETOOLONG, with: :log_image_data_to_datadog
+  rescue_from Errno::ENAMETOOLONG, with: :handle_name_too_long
 
   def create
     rate_limit!(:organization_creation)
-
-    @tab = "organization"
-    @user = current_user
-    @tab_list = @user.settings_tab_list
-
-    unless valid_image?
-      render template: "users/edit"
-      return
-    end
-
-    @organization = Organization.new(organization_params)
+    set_user_info
+    @form = OrganizationForm.new(organization_attributes: organization_params, current_user: current_user)
+    @organization = @form.organization
     authorize @organization
```

Comment:
```suggestion
    authorize @form
```

and you'll need to add

```ruby
def model_name
  Organization.model_name
end
```
to the form object